autometa/config/databases.py

```diff
@@ -245,31 +245,37 @@ def format_nr(self):
         db_infpath = self.config.get("ncbi", "nr")
         db_infpath_md5 = f"{db_infpath}.md5"
         db_outfpath = db_infpath.replace(".gz", ".dmnd")
+
+        db_outfpath_exists = os.path.exists(db_outfpath)
+        if db_outfpath_exists:
+            db_outfpath_hash, __ = calc_checksum(db_outfpath).split()
+
         checksums_match = False
-        if os.path.exists(db_infpath_md5) and os.path.exists(db_outfpath):
-            # If nr.dmnd.md5 exists, then we will check if nr.gz.md5 is
-            # matching
-            current_checksum = read_checksum(db_infpath_md5)
-            current_hash, __ = current_checksum.split()
-            remote_checksum = self.get_remote_checksum("ncbi", "nr")
-            remote_hash, __ = remote_checksum.split()
+        # Check database and database checksum is up-to-date
+        if os.path.exists(db_infpath_md5) and db_outfpath_exists:
+            # Check if the current db md5 is up-to-date with the remote db md5
+            current_hash, __ = read_checksum(db_infpath_md5).split()
+            remote_hash, __ = self.get_remote_checksum("ncbi", "nr").split()
             if current_hash == remote_hash:
-                logger.debug("nr checksums match, skipping...")
+                checksums_match = True
+            # Check if the current db md5 matches the calc'd db checksum
+            if db_outfpath_hash == current_hash:
                 checksums_match = True
 
         dmnd_md5 = f"{db_outfpath}.md5"
-        if os.path.exists(dmnd_md5) and os.path.exists(db_outfpath):
-            dmnd_md5_hash = read_checksum(dmnd_md5).split()[0]
-            db_outfpath_hash = calc_checksum(db_outfpath).split()[0]
-            if dmnd_md5_hash != db_outfpath_hash:
-                write_checksum(db_outfpath, dmnd_md5)
+        if os.path.exists(dmnd_md5) and db_outfpath_exists:
+            dmnd_md5_hash, __ = read_checksum(dmnd_md5).split()
+            if db_outfpath_hash == dmnd_md5_hash:
+                checksums_match = True
```

Comment:
`checksums_match` should be set to `False` before this block, because it is likely to have been set to `True` further up. Therefore, in the failure case where `db_outfpath_hash != dmnd_md5_hash`, it will still be `True`.